packages/source-fetcher/lib/sourcify.ts

```diff
@@ -0,0 +1,136 @@
+import debugModule from "debug";
+const debug = debugModule("source-fetcher:sourcify");
+
+import { Fetcher, FetcherConstructor } from "./types";
+import * as Types from "./types";
+import { removeLibraries } from "./common";
+import request from "request-promise-native";
+
+//this looks awkward but the TS docs actually suggest this :P
+const SourcifyFetcher: FetcherConstructor = class SourcifyFetcher
+  implements Fetcher {
+  get fetcherName(): string {
+    return "sourcify";
+  }
+  static get fetcherName(): string {
+    return "sourcify";
+  }
+
+  static async forNetworkId(
+    id: number,
+    options?: Types.FetcherOptions
+  ): Promise<SourcifyFetcher> {
+    //in the future, we may add protocol and node options,
+    //but these don't exist yet
+    return new SourcifyFetcher(id);
+  }
+
+  private networkId: number;
+
+  private domain: string = "contractrepo.komputing.org";
+
+  constructor(networkId: number) {
+    //we'll just treat all network IDs as valid;
+    //that wouldn't work very well if we were using the IPFS interface,
+    //which goes by network name, but it'll work fine for the HTTP interface
+    //(no idea why they're different)
+    //(technically it should be chain ID but realistically it won't matter)
+    this.networkId = networkId;
+  }
+
+  async isNetworkValid(): Promise<boolean> {
```

Comment:
A little confused what this function is doing - it just always returns true and I don't see it being used anywhere? I see above you say "we'll treat al network IDs as valid", so is this just setting up for any network id validity checks in the future?

Reply:
Basically, yeah.  Both the future if I decide to add that check here -- and it's possible I really should add that check here now, as it could save an HTTPs request, although realistically it's not that likely to come up -- and also the past.  The idea is that all the fetchers adhere to a uniform interface, as given in [types.ts](https://github.com/trufflesuite/truffle/blob/develop/packages/source-fetcher/lib/types.ts).  If you look at the existing [Etherscan fetcher](https://github.com/trufflesuite/truffle/blob/develop/packages/source-fetcher/lib/etherscan.ts), you'll see it does actually potentially return false!  So like I said, the future but also the past. :)

Reply:
OK @gnidan seems to agree I should make this change, so I'll do it. :P